chainstate/test-framework/src/test_block_info.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::TestChainstate;
+use chainstate::chainstate_interface::ChainstateInterface;
+use common::{
+    chain::{gen_block::GenBlockId, Block, GenBlock, Genesis, OutPointSourceId, TxOutput},
+    primitives::{Id, Idable},
+};
+
+// TODO: Replace by a proper UTXO set abstraction
```

Comment:
This is still not achievable with the current changes?

Reply:
We have an issue for this: https://github.com/mintlayer/mintlayer-core/issues/312
Perhaps @iljakuklic can provide more details if someone wants to improve this part.